# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1722169942 -7200
# Parent  4554ac673cdfc3cdb2a8a26660ef0e5fee212e85
Improve QuickFilter status feedback: Fix weird issue in 115 where characters are gobbled in QF input field.

diff --git a/mail/base/content/quickFilterBar.js b/mail/base/content/quickFilterBar.js
--- a/mail/base/content/quickFilterBar.js
+++ b/mail/base/content/quickFilterBar.js
@@ -459,9 +459,9 @@ var quickFilterBar = {
    * Update the UI to reflect the state of the filterer constraints.
    *
    * @param [aFilterName] If only a single filter needs to be updated, name it.
    */
-  reflectFiltererState(aFilterName) {
+  reflectFiltererState(aFilterName, hack = false) {
     // If we aren't visible then there is no need to update the widgets.
     if (this.filterer.visible) {
       let filterValues = this.filterer.filterValues;
       for (let filterDef of QuickFilterManager.filterDefs) {
@@ -477,9 +477,9 @@ var quickFilterBar = {
         if (!("reflectInDOM" in filterDef)) {
           domNode.pressed = value === false || value === true;
           domNode.inverted = value === false;
         } else {
-          filterDef.reflectInDOM(domNode, value, document, this);
+          if (!hack) filterDef.reflectInDOM(domNode, value, document, this);
         }
       }
     }
 
@@ -658,9 +658,9 @@ var quickFilterBar = {
 
   /**
    * Called by the view wrapper so we can update the results count.
    */
-  onMessagesChanged() {
+  onMessagesChanged(hack = false) {
     let filtering = gViewWrapper?.search?.userTerms != null;
     let newCount = filtering ? gDBView?.numMsgsInView : null;
     this.filterer.setFilterValue("results", newCount, true);
 
@@ -691,9 +691,9 @@ var quickFilterBar = {
       }
     }
 
     // - Update match status.
-    this.reflectFiltererState();
+    this.reflectFiltererState(undefined, hack);
   },
 
   /**
    * The displayed folder changed. Reset or reapply the filter, depending on
diff --git a/mail/modules/QuickFilterManager.jsm b/mail/modules/QuickFilterManager.jsm
--- a/mail/modules/QuickFilterManager.jsm
+++ b/mail/modules/QuickFilterManager.jsm
@@ -1544,9 +1544,9 @@ QuickFilterManager.defineFilter({
     if (aStatus != 0) {
       // Interrupted.
       return [null, true, false];
     }
-    aMuxer.onMessagesChanged();
+    aMuxer.onMessagesChanged(true);
     return [scratch.count, false, false];
   },
 
   reflectInDOM(aNode, aFilterValue, aDocument) {
